File: spring/controller-advice-demo/src/main/java/dompoo/controller_advice_demo/exception/ErrorResponse.java
package dompoo.controller_advice_demo.exception;

import lombok.Getter;

@Getter
public class ErrorResponse {
    
    private final ErrorType errorType;
    private final String message;
    
    private ErrorResponse(ErrorType errorType, String message) {
        this.errorType = errorType;
        this.message = message;
    }
    
    public static ErrorResponse fromErrorType(ErrorType errorType) {
        return new ErrorResponse(errorType, errorType.getMessage());
    }
}

File: 언어/kopring/src/main/kotlin/dompoo/kopringdemo/api/exception/util/ErrorResponse.kt
package dompoo.kopringdemo.api.exception.util

data class ErrorResponse(
	val code: Int,
	val message: String,
) {
	companion object {
		fun from(exception: CustomException): ErrorResponse = ErrorResponse(
			code = exception.code,
			message = exception.message ?: ""
		)
	}
}